src/fragments/lib/datastore/native_common/data-access.mdx

```diff
@@ -44,10 +44,26 @@ import flutter8 from "/src/fragments/lib/datastore/flutter/data-access/update-sn
 
   **Avoid working with stale data!**
 
-  Model instances become stale when the original records are overwritten. Saving a record from a copy of a stale model instance can produce errors and/or unexpected states. Fetch records and make fresh copies as close to saving as possible. This can be aided by [observing](/lib/datastore/real-time) a model instance for changes.
+  Model instances which store values, such as those from the results of a `DataStore.Query` operation, can become stale and outdated when their properties are updated. This can be the result of a manual change or even a side effect of [real time data](/lib/datastore/sync#distributed-data) being received by the application. In order to ensure you are performing mutations on the latest committed state to the system, either perform a query directly before the `DataStore.save()` operation or observe the model to keep the state updated at all times and perform mutations on that latest state referencing the model instance. The preceding example demonstrates one approach. The following example demonstrates the `observeQuery` approach.
```

Comment:
this makes sense, if the local data in memory is up-to-date because the app is subscribed to the local store via observe or observeQuery API then the local data will always be the latest before DataStore.save().